service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/ControlledAzureDatabaseResource.java

```diff
@@ -0,0 +1,317 @@
+            flightBeanBag.getCrlService(),
+            this,
+            flightBeanBag.getLandingZoneApiDispatch(),
+            flightBeanBag.getSamService(),
+            flightBeanBag.getWorkspaceService(),
+            getWorkspaceId()),
+        RetryRules.cloud());
+  }
+
+  // Azure resources currently do not implement updating.
+  @Override
+  public void addUpdateSteps(UpdateResourceFlight flight, FlightBeanBag flightBeanBag) {}
+
+  public ApiAzureDatabaseResource toApiResource() {
+    return new ApiAzureDatabaseResource()
+        .metadata(super.toApiMetadata())
+        .attributes(toApiAttributes());
+  }
+
+  private ApiAzureDatabaseAttributes toApiAttributes() {
+    return new ApiAzureDatabaseAttributes().databaseName(getDatabaseName());
+  }
+
+  @Override
+  public String attributesToJson() {
+    return DbSerDes.toJson(
+        new ControlledAzureDatabaseAttributes(
+            getDatabaseName(), getDatabaseOwner(), k8sNamespace, getRegion()));
+  }
+
+  @Override
+  public ApiResourceAttributesUnion toApiAttributesUnion() {
+    ApiResourceAttributesUnion union = new ApiResourceAttributesUnion();
+    union.azureDatabase(toApiAttributes());
+    return union;
+  }
+
+  @Override
+  public void validate() {
+    super.validate();
+    if (getResourceType() != WsmResourceType.CONTROLLED_AZURE_DATABASE
+        || getResourceFamily() != WsmResourceFamily.AZURE_DATABASE
+        || getStewardshipType() != StewardshipType.CONTROLLED) {
+      throw new InconsistentFieldsException("Expected controlled AZURE_DATABASE");
+    }
+    if (getDatabaseName() == null) {
+      throw new MissingRequiredFieldException(
+          "Missing required databaseName field for ControlledAzureDatabase.");
+    }
+    if (getDatabaseOwner() == null) {
+      throw new MissingRequiredFieldException(
+          "Missing required databaseOwner field for ControlledAzureDatabase.");
+    }
+    if (getRegion() == null) {
+      throw new MissingRequiredFieldException(
+          "Missing required region field for ControlledAzureDatabase.");
+    }
+    AzureResourceValidationUtils.validateAzureDatabaseName(getDatabaseName());
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    if (!super.equals(o)) return false;
+
+    ControlledAzureDatabaseResource that = (ControlledAzureDatabaseResource) o;
+
+    return databaseName.equals(that.getDatabaseName());
+  }
+
+  @Override
+  public boolean partialEqual(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    if (!super.partialEqual(o)) return false;
+
+    ControlledAzureDatabaseResource that = (ControlledAzureDatabaseResource) o;
+
+    return databaseName.equals(that.getDatabaseName());
+  }
+
+  @Override
+  public int hashCode() {
+    int result = super.hashCode();
+    result = 31 * result + databaseName.hashCode();
```

Comment:
The database name is enough to uniquely identify it? I would think a combination of the k8snamespace and the database name.

Reply:
the database does not exist in a namespace, the namespace is used to setup the federated identity and k8s service account